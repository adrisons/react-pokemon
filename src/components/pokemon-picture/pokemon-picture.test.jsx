import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Picture from "./pokemon-picture";

describe("WHEN: Picture", () => {
  const component = renderer.create(
    <Router>
      <Picture id={1} />
    </Router>
  );
  const tree = component.toJSON();
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  it("THEN: img must have src with pokemon id url and alt = Pokemon", () => {
    render(
      <Router>
        <Picture id={1} />
      </Router>
    );
    const img = screen.getByAltText("Pokemon");
    expect(img).toHaveAttribute(
      "src",
      "https://pokeres.bastionbot.org/images/pokemon/1.png"
    );
    expect(img).toHaveAttribute("alt", "Pokemon");
  });
});
