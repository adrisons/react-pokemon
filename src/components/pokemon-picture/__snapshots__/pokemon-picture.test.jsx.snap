// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN: Picture THEN: should match snapshot 1`] = `
<div
  className="picture-container"
>
  <div
    className="loading"
  >
    <img
      alt="loading"
      src="pokeball.png"
    />
  </div>
  <img
    alt="Pokemon"
    className="pokemon-picture"
    onLoad={[Function]}
    src="https://pokeres.bastionbot.org/images/pokemon/1.png"
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
